File: packages/python/elizaos/features/basic_capabilities/actions/ignore.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import TYPE_CHECKING

from elizaos.generated.spec_helpers import require_action_spec
from elizaos.types import Action, ActionExample, ActionResult
from elizaos.utils.spec_examples import convert_spec_examples

if TYPE_CHECKING:
    from elizaos.types import HandlerCallback, HandlerOptions, IAgentRuntime, Memory, State

# Get text content from centralized specs
_spec = require_action_spec("IGNORE")


def _convert_spec_examples() -> list[list[ActionExample]]:
    """Convert spec examples to ActionExample format."""
    return convert_spec_examples(_spec)


@dataclass
class IgnoreAction:
    name: str = _spec["name"]
    similes: list[str] = field(default_factory=lambda: list(_spec.get("similes", [])))
    description: str = _spec["description"]

    async def validate(
        self, runtime: IAgentRuntime, _message: Memory, _state: State | None = None
    ) -> bool:
        return True

    async def handler(
        self,
        runtime: IAgentRuntime,
        message: Memory,
        state: State | None = None,
        options: HandlerOptions | None = None,
        callback: HandlerCallback | None = None,
        responses: list[Memory] | None = None,
    ) -> ActionResult:
        if callback and responses and len(responses) > 0:
            first_response = responses[0]
            if first_response.content:
                await callback(first_response.content)

        return ActionResult(
            text="Ignoring message",
            values={"success": True, "ignored": True},
            data={"actionName": "IGNORE"},
            success=True,
        )

    @property
    def examples(self) -> list[list[ActionExample]]:
        return _convert_spec_examples()


ignore_action = Action(
    name=IgnoreAction.name,
    similes=IgnoreAction().similes,
    description=IgnoreAction.description,
    validate=IgnoreAction().validate,
    handler=IgnoreAction().handler,
    examples=IgnoreAction().examples,
)

File: packages/typescript/src/features/basic-capabilities/actions/ignore.ts
import { requireActionSpec } from "../../../generated/spec-helpers.ts";
import type {
	Action,
	ActionExample,
	HandlerCallback,
	HandlerOptions,
	IAgentRuntime,
	Memory,
	State,
} from "../../../types/index.ts";

// Get text content from centralized specs
const spec = requireActionSpec("IGNORE");

export const ignoreAction: Action = {
	name: spec.name,
	similes: spec.similes ? [...spec.similes] : [],
	validate: async (_runtime: IAgentRuntime, _message: Memory) => {
		return true;
	},
	description: spec.description,
	handler: async (
		_runtime: IAgentRuntime,
		_message: Memory,
		_state?: State,
		_options?: HandlerOptions,
		callback?: HandlerCallback,
		responses?: Memory[],
	) => {
		if (callback && responses?.[0]?.content) {
			await callback(responses[0].content);
		}
		return {
			text: "",
			values: { success: true, ignored: true },
			data: { actionName: "IGNORE" },
			success: true,
		};
	},
	examples: (spec.examples ?? []) as ActionExample[][],
} as Action;
